L1Trigger/L1THGCal/test/HGCalTriggerThresholdTriggerCellTester.cc

```diff
@@ -0,0 +1,525 @@
+        triggerCellSimHits_noThreshold_  = fs_->make<TH1F>("triggerCellSimHits_noThreshold","Trigger cell simhit energies, no best choice", 500, 0, 0.16);
+    }
+    triggerCellData_Threshold_ = fs_->make<TH1F>("triggerCellData_Threshold_","Trigger cell values, best choice", 1000, 0., 70000.);
+    if (is_Simhit_comp_) triggerCellData_Threshold_vsSimHits_ = fs_->make<TH2F>("triggerCellData_Threshold_vsSimHits_","Trigger cell values vs simhit energies, best choice", 500,0,0.16,1000, 0., 70000.);
+    triggerCellData_  = fs_->make<TH1F>("triggerCellData","Trigger cell values", 1100, 0., 1100.);
+    if (is_Simhit_comp_)
+    {
+        triggerCellData_vsSimHits_  = fs_->make<TH2F>("triggerCellData_vsSimHits_","Trigger cell values vs simhit energies", 500,0,0.16,1100, 0., 1100.);
+    }
+    triggerCellsPerModule_ = fs_->make<TH1F>("triggerCellsPerModule","Number of trigger cells per module", 64, 0., 64.);
+    triggerCellModuleSum_ = fs_->make<TH1F>("TriggerCellModuleSum","Trigger cell sum in modules", 1000, 0., 10000.);
+    //
+    selectedCellsVsAllCells_ee_ = fs_->make<TH2F>("selectedCellsVsAllCells_ee","Number of selected cells vs number of cell", 128, 0, 128, 128, 0., 128.);
+    energyLossVsNCells_ee_ = fs_->make<TH2F>("energyLossVsNCells_ee","Relative energy loss after selection vs number of cell", 128, 0., 128., 101, 0, 1.01);
+    selectedCellsVsAllCells_fh_ = fs_->make<TH2F>("selectedCellsVsAllCells_fh","Number of selected cells vs number of cell", 128, 0, 128, 128, 0., 128.);
+    energyLossVsNCells_fh_ = fs_->make<TH2F>("energyLossVsNCells_fh","Relative energy loss after selection vs number of cell", 128, 0., 128., 101, 0, 1.01);
+
+}
+
+
+
+HGCalTriggerThresholdTriggerCellTester::~HGCalTriggerThresholdTriggerCellTester() 
+{
+}
+
+void HGCalTriggerThresholdTriggerCellTester::beginRun(const edm::Run& /*run*/, 
+        const edm::EventSetup& es)
+{
+    es.get<IdealGeometryRecord>().get(triggerGeometry_);
+
+    const std::string& ee_sd_name = triggerGeometry_->eeSDName();
+    const std::string& fh_sd_name = triggerGeometry_->fhSDName();
+    const std::string& bh_sd_name = triggerGeometry_->bhSDName();
+    es.get<IdealGeometryRecord>().get(ee_sd_name,info_.geom_ee);
+    es.get<IdealGeometryRecord>().get(fh_sd_name,info_.geom_fh);
+    es.get<IdealGeometryRecord>().get(bh_sd_name,info_.geom_bh);
+    es.get<IdealGeometryRecord>().get(ee_sd_name,info_.topo_ee);
+    es.get<IdealGeometryRecord>().get(fh_sd_name,info_.topo_fh);
+    es.get<IdealGeometryRecord>().get(bh_sd_name,info_.topo_bh);
+
+}
+
+void HGCalTriggerThresholdTriggerCellTester::analyze(const edm::Event& e, 
+        const edm::EventSetup& es) 
+{
+    checkSelectedCells(e, es);
+    rerunThresholdFragments(e, es);
+
+}
+
+void HGCalTriggerThresholdTriggerCellTester::checkSelectedCells(const edm::Event& e, 
```

Comment:
This function in the best choice tester has received updates that should be propagated here.